### PR TITLE
Add a plugin that hides server avatars

### PR DESCRIPTION
**Pull request by crwn1337**
There's probably a better way to do this, but it works for now.

### CODE REVIEW

File: src/plugins/hideServerAvatars/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+
+export default definePlugin({
+    name: "HideServerAvatars",
```
Comment (Vendicated) [inline src/plugins/hideServerAvatars/index.ts:23]:
```suggestion
    name: "NoServerAvatars",
```

File: src/plugins/noServerAvatars/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            replacement: {
+                match: /this\.guildMemberAvatars\[\w+\]/,
+                replace: "null",
+            },
+        },
+        // for replies
+        {
+            find: "renderSingleLineMessage:",
+            replacement: {
+                match: /t\.guildMemberAvatar/,
```
Comment (Vendicated) [inline src/plugins/noServerAvatars/index.ts:39]:
```suggestion
                match: /\i\.guildMemberAvatar/,
```

File: src/plugins/noServerAvatars/index.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+export default definePlugin({
+    name: "NoServerAvatars",
+    description: "Disables server avatars.",
+    authors: [Devs.crwn],
+    patches: [
+        // for chat messages
+        {
+            find: ".default.getGuildMemberAvatarURLSimple",
+            replacement: {
+                match: /this\.guildMemberAvatars\[\w+\]/,
```
Comment (Vendicated) [inline src/plugins/noServerAvatars/index.ts:31]:
```suggestion
                match: /this\.guildMemberAvatars\[\i]/,
```

File: src/plugins/noServerAvatars/index.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+
+export default definePlugin({
+    name: "NoServerAvatars",
+    description: "Disables server avatars.",
```
Comment (Vendicated) [inline src/plugins/noServerAvatars/index.ts:24]:
```suggestion
    description: "Disables server avatars",
```

**Review by Vendicated — COMMENTED** [review]
looks good mostly. could you please test if this still works?

**Comment by crwn1337** [comment]
tested and should be ready to merge

**Pull request closed without merging**